pkg/apis/planetscale/v2/etcdlockserver_types.go

```diff
@@ -197,6 +197,12 @@ type EtcdLockserverTemplate struct {
 	// +kubebuilder:validation:Schemaless
 	// +kubebuilder:pruning:PreserveUnknownFields
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// TopologySpreadConstraint can optionally be used to
+	// specify how to spread etcd pods among the given topology
+	// +kubebuilder:validation:Schemaless
+	// +kubebuilder:pruning:PreserveUnknownFields
+	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
```

Comment:
Any reason why `TopologySpreadConstraint` was added to only a few resources (`etcd`, `VitessDashboard` ,`VtAdmin`, `VtOrc`) ?

Reply:
Those were the only resources that didn't have it already.

It was implemented previously in:
* VitessCell (configures VtGate resources): https://github.com/planetscale/vitess-operator/blob/1e0acc1444d166434da2d822e9232e57f78324f0/pkg/apis/planetscale/v2/vitesscell_types.go#L202
* VitessShard (configures VtTablet resources): https://github.com/planetscale/vitess-operator/blob/1e0acc1444d166434da2d822e9232e57f78324f0/pkg/apis/planetscale/v2/vitessshard_types.go#L277

I think with these changes we are covering all of them now. Unless I've missed something? 🤔